### ISSUE TITLE
[Week 4] (HIGH) Yield PnL accounting is inverted, overstating losses and understating gain

### ISSUE THREAD

**Issue by Emmanuel-hub321** [Frankcastleauditor/Solana-Audit-Arena#213]
Description
`withdraw_from_yield` computes the actual vault delta after withdrawing from `yield_generator`:
```rust
let total_return = checked_sub_u64(vault_after, vault_before)?;
let pnl_i128 = (total_return as i128) - (principal_returned as i128);
```
The intended accounting, documented in `FrankSol/README.md` and `PROGRAM_GUIDE.md`, is:

positive PnL increases `pool.total_sol`
negative PnL decreases `pool.total_sol`
However, the implementation does the opposite:
```rust
if pnl_i128 >= 0 {
    pool.total_sol = PodU64::from(checked_sub_u64(pool.total_sol.get(), gain)?);
} else {
    pool.total_sol = PodU64::from(checked_add_u64(pool.total_sol.get(), loss)?);
}
```
Positive yield decreases pool assets, while losses increase them.

Impact
This corrupts the `frankSOL` exchange rate.

In a negative-yield scenario, the pool records more SOL than it actually controls. Early redeemers can withdraw more than their fair share, leaving later frankSOL holders under-backed or insolvent.

In a positive-yield scenario, the protocol under-credits yield to holders and may strand value.

Proof of Concept
Pool has:
`total_sol `= 100 SOL
`franksol_supply `= 100 frankSOL
Fund manager deploys 100 SOL to yield.
Yield strategy returns only 90 SOL.
The actual PnL is -10 SOL.
Correct result: `pool.total_sol` = 90 SOL.
Actual result: code adds the loss, setting `pool.total_sol `= 110 SOL.
Now `frankSOL` is priced as if the pool has 110 SOL, while only 90 SOL is liquid. Early unstakers can drain the remaining vault balance at an inflated exchange rate.

Recommendation
Reverse the accounting branches:
```rust
if total_return >= principal_returned {
    let gain = checked_sub_u64(total_return, principal_returned)?;
    pool.total_sol = PodU64::from(checked_add_u64(pool.total_sol.get(), gain)?);
} else {
    let loss = checked_sub_u64(principal_returned, total_return)?;
    pool.total_sol = PodU64::from(checked_sub_u64(pool.total_sol.get(), loss)?);
}
```

**Comment by novoyd** [comment]
Duplicate of #144/#146/#175/#186. Same inverted PnL accounting in `withdraw_from_yield`.

**Comment by Frankcastleauditor** [comment]
**Verdict: ❌ DUPLICATE of #144**
**Score: 0 points**
